vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/ApolloWSHandler.java

```diff
@@ -68,6 +71,17 @@ static ApolloWSHandler create(GraphQL graphQL, ApolloWSOptions options) {
   @Fluent
   ApolloWSHandler connectionHandler(Handler<ServerWebSocket> connectionHandler);
 
+  /**
+   * Customize the connection params {@link Handler}.
+   * This handler will be called at CONNECTION_INIT message.
+   * Failed promise or promise resolved to null will result in denied connection, if promise completes
+   * with the object then resulting object will be available at {@link ApolloWSMessage#connectionParams()}.
+   *
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  ApolloWSHandler connectionParamsHandler(BiConsumer<JsonObject, Promise<Object>> connectionParamsHandler);
```

Comment:
`BiConsumer` is not going to work with Vert.x Codegen (which is used to create the Kotlin, Groovy, Scala, Rx APIs).

Instead of adding a new method here, we could add a method on `ApolloWSMessage`:

```java
void setHandshake(Future<Object> future);
```

This is similar to what we have in the websocket API.

A few rules:

- this method should be called only when message type is `CONNECTION_INIT`, otherwise it should throw an `IllegalStateException`
- if the user invokes this method with a non-_null_ instance, the Apollow handler implementation does not reply to the client until the future completes